icu4c/source/i18n/number_stringbuilder.cpp

```diff
@@ -497,4 +514,26 @@ bool NumberStringBuilder::containsField(Field field) const {
     return false;
 }
 
+bool NumberStringBuilder::trimFieldPosition(UFieldPosition& fp) const {
```

Comment:
This has to be UFieldPosition instead of FieldPosition because the former is POD, whereas the latter has a virtual destructor that is implemented in a big dependency object file.